week-2-wide-perspective/w02e04-dungeons-and-the-princess/src/Underground.ts

```diff
@@ -0,0 +1,44 @@
+import { Person } from './Person.js'
+
+export class Underground {
+    private static INITIAL_NO_OF_BARRICADES = 3
+    private noOfBarricades = Underground.INITIAL_NO_OF_BARRICADES
+
+    constructor(private prisoner: Person) {}
+
+    enter(savior: Person) {
+        if (savior.title === 'king') {
+            throw new Error('King cannot enter the underground!')
```

Comment:
Brakuje tutaj dalej jednego elementu.

Gdybyś dla każdego z errorów w zależności od "poziomu" @zawadzki-dawid ustalił klasę, która dziedziczy po `Error` dzięki `instanceof` od razu wiedziałbyś Skąd dokładnie jest `error`

zobacz:

```typescript
class UndergroundError extends Error {}

// a potem dalej w kodzie:

try {

  // .... to co rzuci ten błąd powyżej....

} catch (e: unknown) {

   if(e instanceof UndergroundError) {
       console.log('Error from underground !')
   }
}
```

Podobnie byś to sprawdzał w testach